src/jrd/Mapping.cpp

```diff
@@ -784,13 +784,24 @@ class MappingIpc FB_FINAL : public Firebird::IpcObject
 			iscLogException("MappingIpc: Cannot initialize the shared memory region", ex);
 			throw;
 		}
-		fb_assert(sharedMemory->getHeader()->mhb_header_version == MemoryHeader::HEADER_VERSION);
-		fb_assert(sharedMemory->getHeader()->mhb_version == MAPPING_VERSION);
-
-		Guard gShared(this);
 
 		MappingHeader* sMem = sharedMemory->getHeader();
 
+		if (sMem->mhb_type != SharedMemoryBase::SRAM_MAPPING_RESET ||
+			sMem->mhb_header_version != MemoryHeader::HEADER_VERSION ||
+			sMem->mhb_version != MAPPING_VERSION)
+		{
```

Comment:
Is it safe to access mhb_* members without a mutex acquired? What if shmem is being initialized by another process right at the moment?

Reply:
shmem is not accesible by other processes while it is initialized.
Also, note, on non-Windows - shmem contains mutex that should be acquired - should we access it when shmem binary layout is not known\verified ?
And, finally - former fb_assert() does the same - check before muex is acquired ;)